navigator/backend/src/main/scala/com/digitalasset/navigator/model/PackageState.scala

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.navigator.model
+
+import com.daml.lf.data.Ref
+import com.daml.lf.iface
+
+import scala.collection.immutable.Map
+
+case class PackageState(packages: PackageState.PackageStore) {
+  import PackageState.PackageStore
+  def append(diff: PackageStore): PackageState = {
+    val newPackages = resolveChoicesIn(
+      appendAndResolveRetroactiveInterfaces(diff)
+    )
+    copy(packages = newPackages)
+  }
+
+  private[this] def appendAndResolveRetroactiveInterfaces(diff: PackageStore): PackageStore = {
+    def lookupIf(packageStore: PackageStore, pkId: Ref.PackageId) =
+      packages
```

Comment:
```suggestion
      packageStore
```

Reply:
`packageStore` is the state of the fold; `packages` resets your state back to the beginning. You can test whether this matters by having two retroactive implementations.

You may wish to use the variable name `packages` for `lookupIf` instead so as to _shadow_ the original, much as it's often wise to write `xs.foldLeft(s){(s, ...`, reusing the variable name `s` so as not to accidentally reset to original state.

Reply:
Ah You are right, if I use packages on line 22. There will be an issue if we have more than 1 retroactive implementation in one package.